brainfuck/executor_test.go

```diff
@@ -0,0 +1,56 @@
+package brainfuck
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"testing"
+)
+
+func captureStdout(f func()) string {
+	old := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	f()
+
+	w.Close()
+	os.Stdout = old
+
+	var buf bytes.Buffer
+	io.Copy(&buf, r)
+	return buf.String()
+}
+
+func TestInterpret(t *testing.T) {
+	type args struct {
+		program string
+	}
+	tests := []struct {
+		name     string
+		args     args
+		expected string
+	}{
+		{
+			name:     "Hello world!",
+			args:     args{"++++++++[>++++[>++>+++>+++>+<<<<-]>+>+>->>+[<]<-]>>.>---.+++++++..+++.>>.<-.<.+++.------.--------.>>+.>++."},
+			expected: "Hello World!\n",
+		},
+
+		{
+			name:     "Dirty hello world",
```

Comment:
are you sure that the correct behaviour in this case?

Reply:
I think, yes. Any third party symbols can be used for commenting instructions and formatting.

Reply:
That's why these symbols will be ignored during parsing.

Reply:
then let's write about that rule somewhere... a user should know about it right?

Reply:
Fixed